pkg/gadgets/helpers.go

```diff
@@ -32,6 +42,8 @@ const (
 	PerfBufferPages = 64
 )
 
+var ErrEmptyEvent = errors.New("empty event")
```

Comment:
I'm wondering if we can replace it by returning a `nil` event?

Reply:
I also considered that, but I thought that one normally expects that if `err` is not `nil`, the event is valid, and a `nil` check is not needed:

```golang
	event, err := parseEvent(record.RawSample)
	if err != nil {
		eventCallback(baseEvent(types.Err(err.Error())))
		continue
	}
	if event == nil {
		continue
	}
	eventCallback(*event)
```

WDYT?

Reply:
I think returning nil is better. btw, in this case that check would be removed when we switch to use pointers everywhere for passing events.

Reply:
Applied in https://github.com/inspektor-gadget/inspektor-gadget/pull/1180